From e7f2e8a4f7e4b13e125d12245413d797e6704a89 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Thu, 10 Jul 2014 19:01:52 -0400
Subject: [scsi] fnic: fix broken FIP discovery by initializing multicast address

Message-id: <1405018912-16921-3-git-send-email-cleech@redhat.com>
Patchwork-id: 85676
O-Subject: [RHEL7.1 e-stor PATCH 2/2] fnic: fix broken FIP discovery by initializing multicast address
Bugzilla: 1100078
RH-Acked-by: Andy Grover <agrover@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

 Bug fix is included here:
 http://marc.info/?l=linux-scsi&m=139907829802240&w=2

 Description of that post is:

 [PATCH 4/4] [SCSI]fnic: assign FIP_ALL_FCF_MACS to fcoe_all_fcfs

 There is a bug in VLAN discovery code and it is only affecting
 standalone c-series. Luns are not visible as we are not sending VLAN
 request on valid Mac address.

 Signed-off-by: Hiral Shah <hishah@cisco.com>

Signed-off-by: Chris Leech <cleech@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/fnic/fnic.h b/drivers/scsi/fnic/fnic.h
index 528d43b..a547c3f 100644
--- a/drivers/scsi/fnic/fnic.h
+++ b/drivers/scsi/fnic/fnic.h
@@ -39,7 +39,7 @@
 
 #define DRV_NAME  "fnic"
 #define DRV_DESCRIPTION  "Cisco FCoE HBA Driver"
-#define DRV_VERSION  "1.5.0.45"
+#define DRV_VERSION  "1.5.0.45a"
 #define PFX   DRV_NAME ": "
 #define DFX                     DRV_NAME "%d: "
 
diff --git a/drivers/scsi/fnic/fnic_fcs.c b/drivers/scsi/fnic/fnic_fcs.c
index 1671325..e4f0057 100644
--- a/drivers/scsi/fnic/fnic_fcs.c
+++ b/drivers/scsi/fnic/fnic_fcs.c
@@ -35,7 +35,7 @@
 #include "cq_enet_desc.h"
 #include "cq_exch_desc.h"
 
-static u8 fcoe_all_fcfs[ETH_ALEN];
+static u8 fcoe_all_fcfs[ETH_ALEN] =  FIP_ALL_FCF_MACS;
 struct workqueue_struct *fnic_fip_queue;
 struct workqueue_struct *fnic_event_queue;
 
-- 
1.7.1